tests/DCPS/DomainRange/DomainRangeTest.cpp

```diff
@@ -0,0 +1,398 @@
+#include "MessengerTypeSupportImpl.h"
+#include "DataReaderListenerImpl.h"
+
+#include "dds/DCPS/Service_Participant.h"
+#include "dds/DCPS/Marked_Default_Qos.h"
+#include "dds/DCPS/WaitSet.h"
+
+#include "dds/DCPS/transport/framework/TransportExceptions.h"
+#include "dds/DCPS/transport/framework/TransportRegistry.h"
+
+#include "dds/DdsDcpsInfrastructureC.h"
+#include "dds/DdsDcpsCoreTypeSupportImpl.h"
+#include "dds/DCPS/GuidConverter.h"
+#include "dds/DCPS/SafetyProfileStreams.h"
+
+#include "dds/DCPS/StaticIncludes.h"
+#ifdef ACE_AS_STATIC_LIBS
+#include "dds/DCPS/RTPS/RtpsDiscovery.h"
+#include "dds/DCPS/transport/rtps_udp/RtpsUdp.h"
+#endif
+
+#include "ace/Arg_Shifter.h"
+#include "ace/OS_NS_stdlib.h"
+
+/*
+  NOTE: The messages may not be processed by the reader in this test.
+  This behavior is not an error.
+*/
+
+const int /*DOMAIN_ID = 100,*/ MSGS_PER_WRITER = 10;
+const int TOTAL_WRITERS = 1, TOTAL_READERS = 1;
+
+#define DEFAULT_FLAGS (THR_NEW_LWP | THR_JOINABLE | THR_INHERIT_SCHED)
+
+class WriterTask : public ACE_Task_Base {
+public:
+  WriterTask (const std::string& writer_id,
+              DDS::DataWriter_var writer,
+              int total_readers)
+    : writer_id_(writer_id)
+    , writer_(writer)
+    , total_readers_(total_readers)
+  { }
+
+  int svc()
+  {
+    ACE_DEBUG((LM_DEBUG, "(%P|%t) WriterTask::svc - starting for writer_id: %C\n", writer_id_.c_str()));
+
+    Messenger::MessageDataWriter_var message_writer =
+      Messenger::MessageDataWriter::_narrow(writer_);
+
+    if (!message_writer) {
+      ACE_ERROR_RETURN((LM_ERROR,
+                        ACE_TEXT("ERROR: %N:%l: main() -")
+                        ACE_TEXT(" _narrow failed!\n")),
+                       -1);
+    }
+
+    // Block until Subscriber is available
```

Comment:
Consider using a function like Util::wait_match() here (tests/Utils/StatusMatching.h).

Reply:
Modified the test to use Util::wait_match.